### PR TITLE
dollar-variable-default rule #198

### PR DESCRIPTION
**Pull request by freezy-sk**
WIP #198

### CODE REVIEW

**Comment by kristerkari** [comment]
Ping me when you want me to review this 👍

**Comment by freezy-sk** [comment]
@kristerkari I've added readme file for rule so you can review it right now :)

**Comment by kristerkari** [comment]
Thanks a lot, great job! 

The code looks good and simple, only some small things came to mind related to the unit tests:
- You could add the same test cases using nested selectors (e.g. `.class { a { $var: 10px !default  } }`) to both `accept` and `reject` for both `ignore` option and the rule with default settings. This way we can make sure that the rule also works when the variables are nested.
- You could add a simple `a { color: blue  }` test case to `accept` just to make sure that no one removes the check for declaration starting with `$` in the future.

Also, if you don't mind, you could add this rule to the main README under list of rules: https://github.com/kristerkari/stylelint-scss#list-of-rules

File: src/rules/dollar-variable-default/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  ],
+
+  reject: [
+    {
+      code: "$var: 10px",
+      message: messages.expected,
+      description: "Global variable without !default"
+    },
+    {
+      code: "a { $var: 10px }",
```
Comment (kristerkari) [inline src/rules/dollar-variable-default/__tests__/index.js]: oh yeah, could you add `line` and `column` numbers to the rejected messages just to make sure that the error is shown at the correct place (`$`-sign)?

Like here:
https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/double-slash-comment-inline/__tests__/index.js#L43-L52

**Comment by kristerkari** [comment]
I don't know if you noticed this comment, as Github decided to hide it:
https://github.com/kristerkari/stylelint-scss/pull/199/files/5af02874200008146b84c1b2e25ced82c54432d4#r165815199

Before merging this, I need to revert the changes that drop support for NodeJS 4.x. That's because stylelint has not released a major version yet, and I don't want to release a major version before they do. By reverting the changes, I can release this as part of a minor version release instead.

**Comment by freezy-sk** [comment]
@kristerkari I finished incorporating all of your hints right now

**Comment by freezy-sk** [comment]
@kristerkari you were so fast with that review :D

**Comment by kristerkari** [comment]
Thanks! I'll merge & release new version after I've reverted a couple of commits. 👍

**Comment by freezy-sk** [comment]
thank you :)

**Comment by kristerkari** [comment]
https://github.com/kristerkari/stylelint-scss/releases/tag/2.3.0 🎉

**Pull request merged**